src/ipa/rpi/controller/rpi/clock_recovery.cpp

```diff
@@ -0,0 +1,121 @@
+#include "clock_recovery.h"
+
+#include <cctype>
+#include <chrono>
+#include <fcntl.h>
+#include <list>
+#include <map>
+#include <strings.h>
+#include <unistd.h>
+#include <vector>
+
+#include <libcamera/base/log.h>
+
+#include "sync_status.h"
+
+using namespace std::chrono_literals;
+using namespace RPiController;
+using namespace libcamera;
+
+LOG_DEFINE_CATEGORY(RPiClockRec)
+
+
+float ClockRecovery::slope()
+{
+	//always just add the last two ones, and remove before in adding fucntion
```

Comment:
Not sure what this comment meant. But something saying that this is the slope of the least squares best fit line would be helpful?